src/prefect/client/schemas/objects.py

```diff
@@ -50,6 +49,7 @@
     validate_parent_and_ref_diff,
 )
 from prefect.client.schemas.schedules import SCHEDULE_TYPES
+from prefect.result_records import ResultRecordMetadata
```

Comment:
interestingly, importing this instead of deferring resolves the failure referenced here: https://github.com/PrefectHQ/prefect/issues/15957#issuecomment-2552240650